Hud map- preliminary changes

halephil opened the pull request:
- Changed Object Organization
- Removed Maze.cs constructors
- Added light gameobjects under sun parent object to improve visabillity

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into HUD-Map

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into HUD-Map

Commit:
Game Object Reorginization/Added Sun

Game Object Reorg:
-border walls appear in  the maaze cell they are associated with
-add sun parent game object with two lights to increase visabillity in
maze

Commit:
Game Object Hiearchy

-changed how parent maze game object is set while maze is being drawing

Commit:
Maze base/sub class bug fix

- removed constructors from the maze.cs
- sub classes now handle the intializeMaze() function call to the base
class

Commit:
Merged from master

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into HUD-Map

# Conflicts:
#	VR Maze/Assets/Scenes/FreeRoam.unity

alexweav reviewed (commented):
When I loaded this, the shadows move around really rapidly when the player turns. Not approving or denying this for right now, but is it supposed to do that?

Code changes look good however. Thats my only concern

Commit:
Fixed Lighting issues

-Changes with lighting didn't persist through due to a merge conflict
before the previous PR

Commit:
Lighting Update

-Changed light intensities

Commit:
SpawnManager/SpawnGO/HUDMiniMap Classes

-Spawn Manager Manages the SpawnGO object and allows GO's to be create
in any selected cell
-SpawnGO contains the GO, cell x and cell z location
-Places GO in correct cell
-HUDMiniMap will be responcible for creating minimap to be displayed in
HUD

Commit:
HUDMiniMap

-Removed MiniMap Bug

Commit:
HUDMiniMapGenerator

-Changed name of class
-fixed syntax error

Commit:
HUDMiniMapGenerator

- Removed the create item call

JStuve committed:
Added preface for HUD and components.

JStuve committed:
Add MainPlayer prefab to keep consistancy accros scnees.

JStuve committed:
Added Texture folder that contains a texture for minimap. HUD_Components now has a minimap.

JStuve committed:
Configured the minimap for Freeroam.
